src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/GeneralPartnerServiceTest.java

```diff
@@ -0,0 +1,141 @@
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.limitedpartnershipsapi.exception.ServiceException;
+import uk.gov.companieshouse.limitedpartnershipsapi.mapper.GeneralPartnerMapper;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.LimitedPartnerType;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.dao.GeneralPartnerDao;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.dao.GeneralPartnerDataDao;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.dto.GeneralPartnerDataDto;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.dto.GeneralPartnerDto;
+import uk.gov.companieshouse.limitedpartnershipsapi.repository.GeneralPartnerRepository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.FILING_KIND_GENERAL_PARTNER;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.URL_GET_GENERAL_PARTNER;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.TransactionUtils.buildTransaction;
+
+@ExtendWith(MockitoExtension.class)
+class GeneralPartnerServiceTest {
+
+    private static final String REQUEST_ID = "request123";
+    private static final String USER_ID = "user123";
+    private static final String SUBMISSION_ID = "submission123";
+
+    @InjectMocks
+    GeneralPartnerService generalPartnerService;
+    @Mock
+    GeneralPartnerRepository repository;
+    @Mock
+    GeneralPartnerMapper mapper;
+    @Mock
+    TransactionService transactionService;
+    @Captor
+    private ArgumentCaptor<GeneralPartnerDao> submissionCaptor;
+
+    @Test
+    void testCreateGeneralPartnerReturnsSuccess() throws ServiceException {
+
+        GeneralPartnerDto dto = createDto();
+        GeneralPartnerDao dao = createDao();
+        dao.setId(SUBMISSION_ID);
+
+        when(mapper.dtoToDao(dto)).thenReturn(dao);
+        when(repository.insert(dao)).thenReturn(dao);
+
+        Transaction testTransaction = buildTransaction();
+
+        String submissionId = generalPartnerService.createGeneralPartner(testTransaction, dto, REQUEST_ID, USER_ID);
+
+        verify(mapper, times(1)).dtoToDao(dto);
+        verify(repository, times(1)).insert(dao);
+        verify(repository, times(1)).save(submissionCaptor.capture());
+
+        GeneralPartnerDao sentSubmission = submissionCaptor.getValue();
+        assertEquals(USER_ID, sentSubmission.getCreatedBy());
+        assertEquals(FILING_KIND_GENERAL_PARTNER, sentSubmission.getData().getKind());
+        assertEquals(SUBMISSION_ID, submissionId);
+        assertEquals(LimitedPartnerType.LEGAL_ENTITY, sentSubmission.getData().getPartnerType());
```

Comment:
References LimtedPartnerType. Also we'll not be storing General Partner Type

Reply:
Was aware, but its the same for both should be something more generic. However, on @mwestacott 's recommendation above, this has been removed.